### PR TITLE
Act 4 - Full usage

### PR DESCRIPTION
**Pull request by skizotenigma**
Added Jamella as working NPC.
Added Halbu as working NPC.
Set Cain to Identify.
Added stash capabilities.

### CODE REVIEW

**Comment by jobithu** [comment]
great work, thanks ! added the A4 part to diablo pull request.

File: src/town/a3.py [hunk 1]
```diff
@@ -44,15 +45,15 @@ def stash_is_open_func():
             found = self._template_finder.search("INVENTORY_GOLD_BTN", img, roi=self._config.ui_roi["gold_btn"]).valid
             found |= self._template_finder.search("INVENTORY_GOLD_BTN", img, roi=self._config.ui_roi["gold_btn_stash"]).valid
             return found
-        if not self._char.select_by_template("A3_STASH", stash_is_open_func, telekinesis=True):
```
Comment (aeon0) [inline src/town/a3.py:47]: It defaults to true

**Review by aeon0 — CHANGES_REQUESTED** [review]
Looks good. Just would be nice if you didnt delete the whole pickit.ini :D Also the walking d2 thing is probably for another change I guess.

**Comment by skizotenigma** [comment]
> 

this is why i had to re do my whole repo. it went ham,

File: src/town/a3.py (hunk 1 above)
Comment (skizotenigma) [reply src/town/a3.py:47]: it's that way in the a5.py. i mostly copied it from there.

File: src/pather.py [hunk 2]
```diff
@@ -214,11 +216,23 @@ def __init__(self, screen: Screen, template_finder: TemplateFinder):
...
+            (Location.A4_TYRAEL_STASH, Location.A4_WP): [161, 160],
+            (Location.A4_TYRAEL_STASH, Location.A4_JAMELLA): [161, 160, 162, 163, 164],
+            (Location.A4_TYRAEL_STASH, Location.A4_HALBU): [161, 160, 162, 163, 164, 165],            
+            (Location.A4_JAMELLA, Location.A4_WP): [164, 163, 162],
+            (Location.A4_JAMELLA, Location.A4_TYRAEL_STASH): [164, 163, 162, 160, 161],            
+            (Location.A4_JAMELLA, Location.A4_HALBU): [165],              
+            (Location.A4_HALBU, Location.A4_WP): [164, 163, 162],
+            (Location.A4_HALBU, Location.A4_TYRAEL_STASH): [164, 163, 162, 160, 161],
+            (Location.A4_HALBU, Location.A4_JAMELLA): [164],        
+            # A3 Town0
```
Comment (aeon0) [inline src/pather.py:235]: Is there more than 1 act 3 town? ; )

File: src/town/a3.py [hunk 3]
```diff
@@ -7,6 +7,7 @@
 from typing import Union
 from template_finder import TemplateFinder
 from utils.misc import wait
+from utils.custom_mouse import mouse
```
Comment (aeon0) [inline src/town/a3.py:10]: whats this import used for? I dont see any changes using the mouse.

File: src/town/a4.py [hunk 4]
```diff
@@ -40,4 +46,41 @@ def wait_for_tp(self) -> Union[Location, bool]:
...
+    def heal(self, curr_loc: Location) -> Union[Location, bool]:
+        if not self._pather.traverse_nodes((curr_loc, Location.A4_JAMELLA), self._char, force_move=True): return False
+        self._npc_manager.open_npc_menu(Npc.JAMELLA)
+        return Location.A4_JAMELLA
+        
+    def open_trade_and_repair_menu(self, curr_loc: Location) -> Union[Location, bool]:
+        if not self._pather.traverse_nodes((curr_loc, Location.A5_HALBU), self._char): return False
+        self._npc_manager.open_npc_menu(Npc.HALBU)
+        self._npc_manager.press_npc_btn(Npc.HALBU, "trade_repair")
+        return Location.A5_HALBU
```
Comment (aeon0) [inline src/town/a4.py:86]: this should be A4_HALBU. Same in L83

File: config/pickit.ini [hunk 5]
```diff
@@ -43,9 +43,8 @@ misc_greater_healing_potion=0
 misc_greater_mana_potion=0
 
 ;misc
-misc_gold=0
```
Comment (aeon0) [inline config/pickit.ini:46]: please revert all pickit.ini changes.

**Review by aeon0 — CHANGES_REQUESTED** [review]

**Review by aeon0 — APPROVED** [review]

**Pull request merged**